Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,271 @@
+        {   
+            string credentialFilepath = "TempFilePath.json";
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, DummyServiceAccountCredentialFileContents);
+
+            var credential = await credentialProvider.GetDefaultCredentialAsync();
+
+            Assert.IsInstanceOf(typeof(ServiceAccountCredential), credential.UnderlyingCredential);
+            Assert.IsTrue(credential.IsCreateScopedRequired);
+
+            var scopes = new[] { "https://www.googleapis.com/auth/cloud-platform" };
+            var scopedCredential = credential.CreateScoped(scopes);
+            Assert.AreNotSame(credential, scopedCredential);
+
+            Assert.IsFalse(scopedCredential.IsCreateScopedRequired);
+            CollectionAssert.AreEqual(scopes, ((ServiceAccountCredential) scopedCredential.UnderlyingCredential).Scopes);
+        }
+
+        #endregion
+
+        #region Invalid Cases
+
+        /// <summary> No credential files or environment variable specified - like a fresh developer's machine.</summary>
+        [Test]
+        public async Task GetDefaultCredential_NoCredentialFiles()
+        { 
+            try
+            {
+                var credential = await credentialProvider.GetDefaultCredentialAsync();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("The Application Default Credentials are not available"));
+            }
+        }
+
+        /// <summary>Environment variable points to a non existant credential file.</summary>
+        [Test]
+        public async Task GetDefaultCredential_MissingCredentialFile()
+        {
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, WellKnownCredentialFilePath);
+            
+            try
+            {
+                var credential = await credentialProvider.GetDefaultCredentialAsync();
+                Assert.Fail();
+            }
+            catch (InvalidOperationException e)
+            {
+                Assert.IsTrue(e.Message.Contains("Please check the value of the Environment Variable GOOGLE_APPLICATION_CREDENTIALS"));
+            }
+        }
+
+        /// <summary>Credential file has invalid content format.</summary>
+        [Test]
+        public async Task GetDefaultCredential_InvalidCredentialFile()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, "Invalid Credentials File Contents");
+
+            try
+            {
+                var credential = await credentialProvider.GetDefaultCredentialAsync();
+                Assert.Fail();
```

Comment:
I think that it will make it easier to read \ understand the output when you run several tests.
I would recommend you adding it, but it's up to you.